db/change_cache.go

```diff
@@ -592,14 +596,16 @@ func (c *changeCache) releaseUnusedSequence(ctx context.Context, sequence uint64
 
 	// Since processEntry may unblock pending sequences, if there were any changed channels we need
 	// to notify any change listeners that are working changes feeds for these channels
+	var channelSet channels.Set
 	changedChannels := c.processEntry(ctx, change)
 	if changedChannels == nil {
-		changedChannels = channels.SetOfNoValidate(unusedSeqChannelID)
+		channelSet = channels.SetOfNoValidate(unusedSeqChannelID)
 	} else {
-		changedChannels.Add(unusedSeqChannelID)
+		channelSet = channels.SetFromArrayNoValidate(changedChannels)
+		channelSet.Add(unusedSeqChannelID)
 	}
-	if c.notifyChange != nil && len(changedChannels) > 0 {
-		c.notifyChange(ctx, changedChannels)
+	if c.notifyChange != nil && len(channelSet) > 0 {
+		c.notifyChange(ctx, channelSet)
 	}
```

Comment:
Are there any cases of `notifyChange` that you don't have to change? You could consider something like this that might? be more readable. If it's not don't worry about it.

Rename `changeCache.notifyChange` on the struct to `changeChange.notifyChangeFunc`

Then create:

```
func (c *changeCache) notifyChange(ctx context.Context, chs []channels.ID) {
    if !c.notifyChangeFunc || len(chs) == 0 {
		return
	}
    c.notifyChangeFunc(ctx,   channels.SetFromArrayNoValidate(chs))
}
```

Reply:
I have renamed to `notifyChangeFunc` I like the suggestion to have this in separate function but there are some areas in unused sequence processing we call notify anyway. I'm not sure if it adds confusion to add this and maybe a separate function for calling whether channels is empty or not.